### PR TITLE
[CameraRoll] Add assetThumbnail prop

### PR DESCRIPTION
**Pull request by Iragne**
in reference with this PR
https://github.com/facebook/react-native/pull/1406

### CODE REVIEW

File: Libraries/Image/Image.ios.js [hunk 1]
```diff
@@ -154,7 +159,11 @@ var Image = React.createClass({
       tintColor: style.tintColor,
     });
     if (isStored) {
-      nativeProps.imageTag = source.uri;
+      if (this.props.assetThumbnail === true){
+        nativeProps.assetThumbnail = source.uri;
+      }else{
```
Comment (brentvatne) [inline Libraries/Image/Image.ios.js]: nit: `} else {`

File: Libraries/Image/Image.ios.js [hunk 2]
```diff
@@ -66,6 +66,11 @@ var Image = React.createClass({
       uri: PropTypes.string,
     }),
     /**
+     * Force display of the ALAsset in thumbnail format 
+     * This property reduce the memory size and only work with ALAsset
+     */
+    assetThumbnail: PropTypes.bool,
```
Comment (brentvatne) [inline Libraries/Image/Image.ios.js]: I think the `assetThumbnail` prop should be moved into the `source` object - `{{uri: '...', thumbnail: true}}`

**Comment by Iragne** [comment]
ok good point ;)

**Comment by Iragne** [comment]
The modification is done

**Comment by Iragne** [comment]
@brentvatne and @vjeux that sound ok for you ?

File: Libraries/Image/Image.ios.js [hunk 3]
```diff
@@ -154,7 +159,11 @@ var Image = React.createClass({
       tintColor: style.tintColor,
     });
     if (isStored) {
-      nativeProps.imageTag = source.uri;
+      if (source.assetThumbnail && source.assetThumbnail === true){
+        nativeProps.assetThumbnail = source.uri;
+      }else{
```
Comment (brentvatne) [inline Libraries/Image/Image.ios.js]: nit: should be `} else {`

File: Libraries/Image/Image.ios.js [hunk 4]
```diff
@@ -178,6 +187,8 @@ var nativeOnlyProps = {
   src: true,
   defaultImageSrc: true,
   imageTag: true,
+  resizeMode: true,
+  assetThumbnail:true,
```
Comment (brentvatne) [inline Libraries/Image/Image.ios.js]: nit: missing a space

File: Libraries/Image/RCTImageLoader.m [hunk 5]
```diff
@@ -72,10 +72,17 @@ + (void)loadImageWithTag:(NSString *)imageTag callback:(void (^)(NSError *error,
         dispatch_async(RCTImageLoaderQueue(), ^{
           // Also make sure the image is released immediately after it's used so it
           // doesn't spike the memory up during the process.
+          
           @autoreleasepool {
+            UIImage *image = nil;
             ALAssetRepresentation *representation = [asset defaultRepresentation];
             ALAssetOrientation orientation = [representation orientation];
-            UIImage *image = [UIImage imageWithCGImage:[representation fullResolutionImage] scale:1.0f orientation:(UIImageOrientation)orientation];
+            if (!thumb){
```
Comment (brentvatne) [inline Libraries/Image/RCTImageLoader.m]: nit: missing space between `)` and `{`

File: Examples/UIExplorer/CameraRollExample.ios.js [hunk 6]
```diff
@@ -70,7 +70,10 @@ var CameraRollExample = React.createClass({
     return (
       <View key={asset} style={styles.row}>
         <Image
-          source={asset.node.image}
+          source={{
+            uri:asset.node.image.uri,
+            assetThumbnail:true
```
Comment (brentvatne) [inline Examples/UIExplorer/CameraRollExample.ios.js]: nit: missing spaces after `:` in both properties

File: Libraries/Image/Image.ios.js (hunk 3 above)
Comment (Iragne) [inline Libraries/Image/Image.ios.js]: ok

File: Examples/UIExplorer/CameraRollExample.ios.js (hunk 6 above)
Comment (Iragne) [inline Examples/UIExplorer/CameraRollExample.ios.js]: ok

Comment (Iragne) [inline Examples/UIExplorer/CameraRollExample.ios.js]: How i can take the "Nit" word ?

**Comment by mrspeaker** [comment]
This patch works like a charm, by the way... really looking forward to it landing!

**Comment by Iragne** [comment]
Prefered resize rather than a native Thumb extraction ;)

**Pull request closed without merging**